### ISSUE TITLE
Hide keyboard approcahces in Flutter. Keyboard reappear even after FocusScope.of(context).unfocus()

### ISSUE THREAD

**Issue by noureen** [flutter/flutter#158365]
I have an issue in understanding of FocusScope.of(context).unfocus() vs FocusManager.instance.primaryFocus?.unfocus() it seems like FocusManager.instance.primaryFocus?.unfocus() is the actual working solution or custom FocusNode is another option which is holding reference of FocusManager.instance.primaryFocus.

** Solution1 which seems to to be working but getting auto focus after any action like after closing bottomsheet**

//solution1
```
final fc = FocusScope.of(context);
        fc.unfocus();
        print(
            "FocusScope Info: ${fc.nearestScope}"); //Info: FocusScopeNode#6e22f(_ModalScopeState<dynamic> Focus Scope [IN FOCUS PATH])(context: FocusScope, IN FOCUS PATH, focusedChildren: [FocusNode#c394a([PRIMARY FOCUS])], traversalEdgeBehavior: TraversalEdgeBehavior.parentScope)
```

** solution2 which seems to actually working**

//solution2
```
        FocusManager.instance.primaryFocus?.unfocus();
        final fcPrimary = FocusScope.of(context);
        print("FocusManager Info: ${fcPrimary.nearestScope}"); //FocusScopeNode#9dfd6(_ModalScopeState<dynamic> Focus Scope [IN FOCUS PATH])(context: FocusScope, IN FOCUS PATH)
```

```
class ChatPage extends StatefulWidget {
  const ChatPage({super.key});

  @override
  State<ChatPage> createState() => _ChatPageState();
}

class _ChatPageState extends State<ChatPage> {
  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: () {

        //solution1
        final fc = FocusScope.of(context);
        fc.unfocus();
        print(
            "FocusScope Info: ${fc.nearestScope}"); //Info: FocusScopeNode#1h34f(_ModalScopeState<dynamic> Focus Scope [IN FOCUS PATH])(context: FocusScope, IN FOCUS PATH, focusedChildren: [FocusNode#c865b([PRIMARY FOCUS])], traversalEdgeBehavior: TraversalEdgeBehavior.parentScope)

        //solution2
        /*FocusManager.instance.primaryFocus?.unfocus();
        final fcPrimary = FocusScope.of(context);
        print("FocusManager Info: ${fcPrimary.nearestScope}");*/ //FocusScopeNode#7hfd3(_ModalScopeState<dynamic> Focus Scope [IN FOCUS PATH])(context: FocusScope, IN FOCUS PATH)
      },
      child: Scaffold(
        body: Stack(
          children: <Widget>[
            Align(
              alignment: Alignment.bottomLeft,
              child: Container(
                padding: const EdgeInsets.only(
                    left: 10, bottom: 10, top: 10, right: 10),
                height: 60,
                width: double.infinity,
                color: Colors.white,
                child: Row(
                  children: <Widget>[
            const SizedBox(
        width: 15,
      ),
                    const Expanded(
                      child: TextField(
                        decoration: InputDecoration(
                            hintText: "Message here...",
                            hintStyle: TextStyle(color: Colors.black54),
                            border: InputBorder.none),
                      ),
                    ),
                    const SizedBox(
        width: 15,
      ),
                    FloatingActionButton(
                      onPressed: () {
                        _buildBottomSheet();
                      },
                      backgroundColor: Colors.blue,
                      elevation: 0,
                      child: const Icon(
                        Icons.share,
                        color: Colors.white,
                        size: 18,
                      ),
                    ),
                  ],
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }



  _openBottomSheet() => showModalBottomSheet(
      context: context,
      builder: (builder) {
        return Container(
          height: 250.0,
          color: Colors.transparent,
          child: Container(
              decoration: const BoxDecoration(
                  color: Colors.white,
                  borderRadius: BorderRadius.only(
                      topLeft: Radius.circular(14.0),
                      topRight: Radius.circular(14.0))),
              child: const Center(
                child: Text("Modal sheet"),
              )),
        );
      });
}
```

**Comment by darshankawar** [comment]
@noureen 
When you trigger showModalBottomSheet, it will open a new route. And it comes with isDismissible as true by default, which means the model is dismissed when tapping outside. This is actually calling Navigator.pop method. If you use Navigator.push to a new page instead when tapping on the avatar, you will also see the same issue.

This is a known issue at https://github.com/flutter/flutter/issues/145155 (or https://github.com/flutter/flutter/issues/48464). You can find a workaround at https://github.com/flutter/flutter/issues/145155#issuecomment-2203536461.

Closing this as a duplicate of these issue. If you disagree, write in comments and I'll reopen it.

**Comment by noureen** [comment]
Thankyou @darshankawar  I understand your point but the thing is when I call
`FocusScope.of(context).unfocus()` and mode to another route and back again keyboard get the focus again event I have called unfocus before that.

By checking other threads I got to know that primaryFocus?.unfocus() is the working solution.
`FocusManager.instance.primaryFocus?.unfocus()`

So my point is why is there a need to mention that `FocusScope.of(context).unfocus() ` can actually hide the keyboard? But in real it is just hiding the keyboard temporarily but keeping the referece of primary node.
